### PR TITLE
AO/MUR document permalinks: Use filename instead of document_ID

### PR DESCRIPTION
**Pull request by lbeaufort**
Use filename to create AO, MUR document pdfs instead of ID, ensuring we have permalinks. Add AO/MUR number to URL IE `legal/aos/2017-12/201712C_3.pdf`

For the legacy files that were created before the new PDF naming convention, I am replacing spaces with "-". I think dashes are be more conventional, but the new naming convention has underscores in them:  "201712C_3.pdf"

*Examples*
```
legal/aos/2012-17/AO-2012-17-Red-Blue-T,-LLC,-ArmourMedia,-Inc.,-and-m-Qube,-Inc.)-Final-(Signed)-(6.11.12).pdf

legal/aos/2017-12/201712C_3.pdf
```

Later need to delete PDFs from `dev`, `stage`, and `prod` S3 with old naming conventions --> Add this to existing cleanup issue. Per @LindsayYoung s3 space is cheap and we don't want to break existing links

- [x] Manually deploy local branch to `dev` and reload all the AO's and MURs, test the new links
- [x] Add URL test to AOs

### CODE REVIEW

File: webservices/legal_docs/current_murs.py [hunk 1]
```diff
@@ -44,11 +44,13 @@
 """
 
 MUR_DOCUMENTS = """
-    SELECT document_id, filename, category, description, ocrtext,
+    SELECT document_id, mur.case_no, filename, category, description, ocrtext,
```
Comment (vrajmohan) [inline webservices/legal_docs/current_murs.py]: Might be a good opportunity to format the SQL like in AOs, one column per line and preceded by the table name/alias

File: webservices/legal_docs/current_murs.py [hunk 2]
```diff
@@ -285,14 +298,16 @@ def get_documents(case_id, bucket, bucket_name):
                 'text': row['ocrtext'],
                 'document_date': row['document_date'],
             }
-            pdf_key = 'legal/murs/current/%s.pdf' % row['document_id']
-            logger.info("S3: Uploading {} Orig filename: {}".format(pdf_key, document['filename']))
+            pdf_key = 'legal/murs/current/{0}/{1}'.format(row['case_no'],
```
Comment (vrajmohan) [inline webservices/legal_docs/current_murs.py]: As discussed offline, we could lose the `/current`. Also, check archived MURs.

File: webservices/legal_docs/advisory_opinions.py [hunk 3]
```diff
@@ -182,12 +185,14 @@ def get_documents(ao_id, bucket):
                 "text": row["ocrtext"],
                 "date": row["document_date"],
             }
-            pdf_key = "legal/aos/%s.pdf" % row["document_id"]
-            logger.info("S3: Uploading {} Orig filename: {}".format(pdf_key, document['filename']))
+            pdf_key = "legal/aos/{0}/{1}".format(row['ao_no'],
+                    str.replace(row["filename"] or '', ' ', '-'))
```
Comment (vrajmohan) [inline webservices/legal_docs/advisory_opinions.py]: As discussed offline, if the `or` is only to handle missing filenames in test data, we can fix the test data.

File: webservices/legal_docs/current_murs.py [hunk 4]
```diff
@@ -285,14 +298,16 @@ def get_documents(case_id, bucket, bucket_name):
...
+                    str.replace(row['filename'] or '', ' ', '-'))
+            document['url'] = '/files/' + pdf_key
+            logger.debug("S3: Uploading {}".format(pdf_key))
```
Comment (vrajmohan) [inline webservices/legal_docs/current_murs.py]: `info` level now and switch to `debug` after everything is working soundly?

File: webservices/legal_docs/current_murs.py (hunk 2 above)
Comment (lbeaufort) [reply webservices/legal_docs/current_murs.py]: @vrajmohan I confirmed archived MURs don't have `/archived` document URLs: https://fec-dev-proxy.app.cloud.gov/files/legal/murs/4772.pdf

File: webservices/legal_docs/advisory_opinions.py [hunk 5]
```diff
@@ -182,12 +185,17 @@ def get_documents(ao_id, bucket):
...
-            logger.info("S3: Uploading {} Orig filename: {}".format(pdf_key, document['filename']))
-            bucket.put_object(Key=pdf_key, Body=bytes(row["fileimage"]),
-                    ContentType="application/pdf", ACL="public-read")
-            document["url"] = '/files/' + pdf_key
-            documents.append(document)
+            if not row['fileimage']:
+                logger.error('Error uploading document ID {0} for AO no {1}: No file image'.format(row['document_id'], row['ao_no']))
+            else:
+                pdf_key = "legal/aos/{0}/{1}".format(row['ao_no'],
+                str.replace(row["filename"], ' ', '-'))
```
Comment (vrajmohan) [inline webservices/legal_docs/advisory_opinions.py]: IMO,
```
row["filename"].replace(' ', '-')
```
would read better, stylistically.

File: webservices/legal_docs/current_murs.py [hunk 6]
```diff
@@ -285,14 +315,20 @@ def get_documents(case_id, bucket, bucket_name):
...
-            logger.info("S3: Uploading {} Orig filename: {}".format(pdf_key, document['filename']))
-            bucket.put_object(Key=pdf_key, Body=bytes(row['fileimage']),
-                    ContentType='application/pdf', ACL='public-read')
-            document['url'] = '/files/' + pdf_key
-            documents.append(document)
+            if not row['fileimage']:
+                logger.error('Error uploading document ID {0} for MUR Case {1}: No file image'.format(row['document_id'], row['case_no']))
+            else:
+                pdf_key = 'legal/murs/{0}/{1}'.format(row['case_no'],
+                str.replace(row['filename'], ' ', '-'))
```
Comment (vrajmohan) [inline webservices/legal_docs/current_murs.py]: See earlier comment

**Comment by lbeaufort** [comment]
@vrajmohan I think this is ready to go

**Pull request merged**